core/cap-0067.md

```diff
@@ -145,14 +151,24 @@ contract: asset, topics: ["clawback", from:Address, sep0011_asset:String], data:
 
 At the moment, if the issuer is the sender in a Stellar Asset Contract `transfer`, the asset will be minted. If the issuer is the recipient, the asset will be burned. The event emitted in both scenarios, however, is the `transfer` event. This CAP changes that behavior to instead emit the `mint`/`burn` event.
 
-### Add a `transfer_with_memo` function to the Stellar Asset Contract
+### Update the `transfer` function to take in a `MuxedAccount` for `from` and `to`
+
+This change will allow the `transfer` event to emit muxed information for exchange support. 
+
+### Update the `data` field in the `transfer` event from an integer to a map to store additional memo information.
 
-Add the following function to the Stellar Asset Contract
-`transfer_with_memo(from: Address, to: Address, amount: i128, from_muxed_id: Option<u64>, to_muxed_id: Option<u64>)` and extend the `transfer` event topics to `["transfer", from: Address, to: Address, sep0011_asset: String, muxed_ids:Vec<Option<U64>>]`.
+Make the `data` field a map on the `transfer` event. Here are the following relevant scenarios and what the resulting events should look like - 
 
-The `transfer_with_memo` function will forward the muxed id's into the new muxed_ids vector on the `transfer` event. The first element will be the muxed id for `from`, and the second element will be the muxed id for `to`. If the optional muxed id is not specified, then a `SCV_VOID` `SCVal` will be emitted instead. The muxed parameters must be passed to the `transfer_with_memo`'s `require_auth` call.
+1. For Soroban transfers -
+   1. If either or both `from` and `to` are of type `KEY_TYPE_MUXED_ED25519`, then the respective `u64` IDs will be pulled out and emitted in the data field - `topics: ["transfer", from:G, to:G] data: { amount: i128, from_muxed_id:u64, to_muxed_id:u64 }`. Note that `from` and `to` are the ed25519 addresses, not the muxed address.
+2. For any of the `transfer` events that can be emitted by non-soroban operations -
+   1. The memo in a non-Soroban transaction can be one of `MEMO_NONE`, `MEMO_TEXT`, `MEMO_ID`, `MEMO_HASH`, or `MEMO_RETURN`. This memo will be forwarded for the destination, so `to_muxed_id` can be one of those types mentioned eg. `topics: ["transfer", from:G, to:G] data: { amount: i128, from_muxed_id:u64, to_muxed_id:u64|String|BytesN<32> }`. Both `MEMO_HASH` and `MEMO_RETURN` use  `BytesN<32>` so the types are indistinguishable in the event.
+   2. If the sender of the `transfer` is muxed, then `from_muxed_id` will be emitted with the `MEMO_ID` on that muxed account.
+   3. If the destination of the `transfer` is muxed, then `to_muxed_id` will be emitted with the `MEMO_ID` on the muxed account. If the destination is not muxed, but the transaction memo is set, then `to_muxed_id` will be emitted with the memo on the transaction.
```

Comment:
Am I understanding correctly that memo info will be encoded in the `to_mux_id` field, so it's sharing the field with the memo ID of the `to` address? Why not make the memo an additional optional field? 
Is it implicit that the `to` address being a muxedAddress and memo field existing cannot be both true?

Reply:
We will choose one based on an order of precedence. We don't want to reintroduce this concept of a transaction memo to soroban, which is why we're going to stop allowing transactions memos in soroban transactions. This order of precedence will only apply for classic transactions.